dotnet.sh

```diff
@@ -23,5 +23,9 @@ source $scriptroot/eng/common/tools.sh
 InitializeDotNetCli true # Install
 __dotnetDir=${_InitializeDotNetCli}
 
+# Temporarily make this dotnet more permanent
```

Comment:
It would be nice to send PR with this change to dotnet/runtime too.

Reply:
This is only needed because we don't publish ilc as a selfcontained app. Do you think it's still valuable in dotnet/runtime?

Reply:
I thought that you have added this for your own convenience to make `dotnet.sh` work on anything. Is it the case, or is this needed to make the CI work as well?

If this is for convenience, I think it would be useful in dotnet/runtime too.
If this is a temporary workaround to make CI work, it would be nice to have better comments on it that describe what it is for and what it would take to remove it.